From 22f8eeee6deddadb8a2027221b97a3b389997871 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 26 Aug 2022 17:09:07 +0300
Subject: [PATCH] [SW-99461] gaudi3: export: add persist data exports

Adding apis to export persist data structures of image descriptors
and security data for re-use in other boot stages, as required.

This is an alternate and also a backup strategy in case boot info
infra is unusable in rom.

Change-Id: I8b91745cb4fb58a9af736fd5d4399a5e8462c820
---

diff --git a/arch/gaudi3/arc/arcboot_rom.txt b/arch/gaudi3/arc/arcboot_rom.txt
index c0ba611..5ac4c4b 100644
--- a/arch/gaudi3/arc/arcboot_rom.txt
+++ b/arch/gaudi3/arc/arcboot_rom.txt
@@ -21,3 +21,5 @@
 security_rom_get_key_hash_by_idx
 security_rom_get_hash
 security_get_rom_key_hash_addr
+arcboot_rom_get_spi_manifest_hdr
+arcboot_rom_get_sec_data
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index d671bf6..c9bf30c 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1142,3 +1142,19 @@
 		boot_info_finalize_no_merge();
 	}
 }
+
+void arcboot_rom_get_spi_manifest_hdr(void *p_dest)
+{
+	if (!p_dest)
+		return;
+
+	memcpy(p_dest, &arcboot_rom_global.spi_manifest_hdr, sizeof(struct hl_fw_boot_ptrs));
+}
+
+void arcboot_rom_get_sec_data(void *p_dest)
+{
+	if (!p_dest)
+		return;
+
+	memcpy(p_dest, &arcboot_rom_global.sec_data, sizeof(struct hl_fw_sec_info));
+}
